airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -470,6 +470,15 @@ public List<CheckedConsumer<JdbcDatabase, Exception>> getCheckOperations(final J
       });
     }
 
+    if (isXmin(config)) {
+      checkOperations.add(database -> {
+        if (PostgresQueryUtils.getXminStatus(database).getNumWraparound() > 0) {
+          throw new ConfigErrorException("The database has experienced transaction ID wraparound. " +
```

Comment:
In many cases - this is an issue that the user cannot solve. I think a better error message would be something along the lines of :

"We detected XMIN transaction wraparound in the source database, which makes this replication method inefficient and can lead to higher credit consumption. Please change the replication method to CDC or cursor based"

Reply:
sounds good, updated!